src/netaddress.cpp

```diff
@@ -5,18 +5,111 @@
 
 #include <netaddress.h>
 
+#include <crypto/sha3.h>
 #include <hash.h>
+#include <prevector.h>
 #include <tinyformat.h>
-#include <util/strencodings.h>
 #include <util/asmap.h>
+#include <util/strencodings.h>
+#include <util/string.h>
 
 #include <algorithm>
 #include <array>
 #include <cstdint>
+#include <ios>
 #include <iterator>
 #include <tuple>
 
 constexpr size_t CNetAddr::V1_SERIALIZATION_SIZE;
+constexpr size_t CNetAddr::MAX_ADDRV2_SIZE;
+
+CNetAddr::BIP155Network CNetAddr::GetBIP155Network() const
+{
+    switch (m_net) {
+    case NET_IPV4:
+        return BIP155Network::IPV4;
+    case NET_IPV6:
+        return BIP155Network::IPV6;
+    case NET_ONION:
+        switch (m_addr.size()) {
+        case ADDR_TORV2_SIZE:
+            return BIP155Network::TORV2;
+        case ADDR_TORV3_SIZE:
+            return BIP155Network::TORV3;
+        default:
+            assert(false);
+        }
+    case NET_I2P:
+        return BIP155Network::I2P;
+    case NET_CJDNS:
+        return BIP155Network::CJDNS;
+    case NET_INTERNAL:   // should have been handled before calling this function
+    case NET_UNROUTABLE: // m_net is never and should not be set to NET_UNROUTABLE
+    case NET_MAX:        // m_net is never and should not be set to NET_MAX
+        assert(false);
+    } // no default case, so the compiler can warn about missing cases
+
+    assert(false);
+}
+
+bool CNetAddr::RecognizeBIP155Network(uint8_t possible_bip155_net, size_t address_size)
```

Comment:
`SetBIP155Network` would be incorrect because `m_net`, which is being set, is of type `Network`, not `BIP155Network`.

Renamed to this function to `SetNetFromBIP155Network()` to imply "set m_net from variable of type BIP155Network".

Reply:
SGTM.